src/module-elasticsuite-catalog/Model/ResourceModel/Product/Indexer/Fulltext/Datasource/AttributeData.php

```diff
@@ -229,6 +245,18 @@ private function getRelationQuery($relation, $parentIds, $storeId)
             )
             ->where("parent.{$entityIdField} in (?)", $parentIds);
 
+        /**
+         * If Catalog - Inventory - Stock Options - Display of Stock Products is set to NO,
+         * then exclude this children from the query results.
+         */
+        if (!$this->scopeConfig->getValue(Configuration::XML_PATH_SHOW_OUT_OF_STOCK)) {
+            $select->joinInner(
+                ['stock' => $inventoryTable],
+                new \Zend_Db_Expr("child.{$entityIdField} = stock.product_id AND stock.is_in_stock = 1"),
+                []
+            );
+        }
+
```

Comment:
Hello @Bashev,

Wouldn't this piece of code only do what it is intended to do if 
* not using MSI
* using MSI but the current storeId corresponds to a store using the default Stock ?

Regards,

Reply:
Probably,
to be honest, i not test it with MSI which have multiple sources and stocks.

I'll try to check these scenarios and will come back.